dashboard/app/assets/javascripts/client_state.js

```diff
@@ -0,0 +1,78 @@
+//= require jquery
+//= require jquery.cookie
+
+(function (window, $) {
+
+if (!window.dashboard) {
+  window.dashboard = {};
+}
+
+dashboard.clientState = {};
+
+/**
+ * Number of days before client state cookie expires.
+ * @type {number}
+ * @private
+ */
+dashboard.clientState.EXPIRY_DAYS = 365;
+
+dashboard.clientState.reset = function() {
+  $.removeCookie('progress');
+  $.removeCookie('lines');
+};
+
+/**
+ * Returns the progress attained for the given level from the cookie.
+ * @param {number} level The id of the level
+ * @returns {number}
+ */
+dashboard.clientState.levelProgress = function(level) {
+  var progressMap = dashboard.clientState.allLevelsProgress();
+  return progressMap[String(level)] || 0;
+};
```

Comment:
The parens end up being hard to read.  I think in this the case two easy-to-read lines are better than one concise but hard to read line

return (dashboard.clientState.allLevelsProgress())[String(level)] || 0;